File: 프로그래머스/2/64065. 튜플/튜플.java
import java.util.*;
class Solution {
    public int[] solution(String s) {
        s = s.substring(2, s.length() - 2);
        String[] sets = s.split("},\\{");

        Arrays.sort(sets, (o1, o2) -> o1.length() - o2.length());
        Set<Integer> ans = new HashSet<>();
        ArrayList<Integer> list = new ArrayList<>();

        for(String str : sets) {
            for(String num : str.split(",")) {
                int n = Integer.parseInt(num);
                if(!ans.contains(n)) {
                    ans.add(n);
                    list.add(n);
                }
            }
        }
        return list.stream().mapToInt(i -> i).toArray();
    }
}

File: 프로그래머스/2/64065. 튜플/튜플.py
def solution(s):
    s = s[2:-2]
    s = s.split('},{')
    s.sort(key=len)
    ans = dict()
    arr = []
    for unit in s:
        l = map(int, unit.split(','))
        for i in l:
            ans[i] = 0
    for k in ans.keys():
        arr.append(k)
    return arr